NodeAFK.js

```diff
@@ -0,0 +1,215 @@
+const desktopIdle = require('desktop-idle');
+const EventEmitter = require('events');
+
+const STATUS_IDLE = 'idle';
+const STATUS_ACTIVE = 'active';
+
+/**
+ * NodeAFK
+ *
+ * @extends EventEmitter
+ */
+class NodeAFK extends EventEmitter {
+  /**
+   * Create a new instance of NodeAFK
+   * @param {number} inactivityDuration The duration until the user is considered `idle` (in ms)
+   * @param {number} pollInterval How often should NodeAFK poll the system to
+   * get the current idle time (in ms)
+   * @param {string} initialStatus The initial status of the user
+   */
+  constructor(
+    inactivityDuration,
+    pollInterval = 1000,
+    initialStatus = STATUS_ACTIVE,
+  ) {
+    super();
+
+    this.inactivityDuration = inactivityDuration;
+    this.pollInterval = pollInterval;
+    this.pollIntervalId = undefined;
+    this.timedEvents = [];
+    this.userLastActiveAt = undefined;
+
+    this.setStatus(initialStatus);
+  }
+
+  /**
+   * Set the current status for the user
+   * @param {string} status The new status for the user
+   * @throws Will throw an error when an invalid status is provided
+   * @public
+   */
+  setStatus(status) {
+    if (![STATUS_IDLE, STATUS_ACTIVE].includes(status)) {
+      throw new Error(`${status} is not a valid status`);
+    }
+
+    this.userLastActiveAt = (status === STATUS_ACTIVE) ? Date.now() : undefined;
```

Comment:
Don't need `()` here, and this should only be set if the user `status === STATUS_ACTIVE` , it doesn't make sense to set it to undefined?

Reply:
Yep theres no reason to set to undefined, we can just update the value when the status changes to `active`